src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,132 @@
 package core.basesyntax;
 
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private int size;
+    private T[] elements;
+
+    public ArrayList() {
+        this(DEFAULT_CAPACITY);
+    }
+
+    public ArrayList(int initialCapacity) {
+        if (initialCapacity < 0) {
+            throw new ArrayListIndexOutOfBoundsException("Illegal Capacity: " + initialCapacity);
+        }
+        this.elements = (T[]) new Object[initialCapacity];
+    }
+
     @Override
     public void add(T value) {
-
+        ensureCapacity(size + 1);
+        elements[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index < 0 || index > size) {
+            throw new ArrayListIndexOutOfBoundsException("Illegal Index: " + index);
+        }
+        ensureCapacity(size + 1);
+        System.arraycopy(this.elements, index, this.elements, index + 1, size - index);
+        this.elements[index] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        int numNew = list.size();
+        T[] arrayList = (T[]) new Object[numNew];
+        for (int i = 0; i < numNew; i++) {
+            arrayList[i] = list.get(i);
+        }
+        System.arraycopy(arrayList, 0, elements, size, numNew);
+        size += numNew;
     }
 
     @Override
     public T get(int index) {
-        return null;
+        if (index < 0 || index >= size) {
+            throw new ArrayListIndexOutOfBoundsException("Illegal Index: " + index);
+        }
```

Comment:
![image](https://user-images.githubusercontent.com/20326858/233032543-61f42907-f19b-4c21-bc4d-24c2ad96f24a.png)

Reply:
not fixed